packages/plugins/input-exercise/src/editor.tsx

```diff
@@ -41,75 +41,65 @@ export function InputExerciseEditor(props: InputExerciseProps) {
+                  <AnswerTextfield
+                    value={answer.value.value}
+                    placeholder={i18n.answer.value.placeholder}
+                    type="text"
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                      answer.value.set(e.target.value)
                     }}
-                    focusedElement={focusedElement || undefined}
                   />
-                )
-              })}
-              <AddButton onClick={() => state.answers.insert()}>
-                {i18n.answer.addLabel}
-              </AddButton>
-            </React.Fragment>
-          ) : null}
-          {props.renderIntoSettings(
-            <React.Fragment>
-              <OverlayInput
-                label={i18n.unit.label}
-                value={state.unit.value}
-                onChange={(e) => {
-                  state.unit.set(e.target.value)
-                }}
+                }
+                feedback={answer.feedback.render()}
+                feedbackID={answer.feedback.id}
+                isActive={answer.isCorrect.value}
+                handleChange={() =>
+                  answer.isCorrect.set(!answer.isCorrect.value)
+                }
+                remove={() => state.answers.remove(index)}
+                focusedElement={focusedElement || undefined}
               />
-              <OverlaySelect
-                label={i18n.type.label}
-                onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
-                  state.type.set(translateDataName(event.target.value))
-                }}
-                value={translateDataType(state.type.value as InputExerciseType)}
-                options={R.values(
-                  R.mapObjIndexed((type) => {
-                    return translateDataType(type)
-                  }, InputExerciseType)
-                )}
-              />
-            </React.Fragment>
-          )}
+            )
+          })}
+          <AddButton onClick={() => state.answers.insert()}>
+            {i18n.answer.addLabel}
+          </AddButton>
+        </React.Fragment>
+      )}
+      {props.renderIntoSettings(
+        <React.Fragment>
+          <OverlayInput
+            label={i18n.unit.label}
+            value={state.unit.value}
+            onChange={(e) => state.unit.set(e.target.value)}
+          />
+          <OverlaySelect
+            label={i18n.type.label}
+            onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+              state.type.set(translateDataName(event.target.value))
+            }}
+            value={translateDataType(state.type.value as InputExerciseType)}
+            options={R.values(
+              R.mapObjIndexed(translateDataType, InputExerciseType)
+            )}
+          />
         </React.Fragment>
-      ) : (
-        <InputExerciseRenderer {...props} />
       )}
     </React.Fragment>
+  ) : (
+    <InputExerciseRenderer {...props} />
```

Comment:
To skip one level of indentation and make it more readable: add that as an early return before the editing part:
```tsx
if (!editable) return <InputExerciseRenderer {...props} />
```